File: 2396. Strictly Palindromic Number/isStrictlyPalindromic.cpp
#include <iostream>
#include <string>
#include <bitset>

using namespace std;

class Solution {
public:
    bool checkPalindrom(string str){
        int left = 0;
        int right = str.size() -1;

        while(left < right){
            if(str[left] != str[right]) return false;
            left++;
            right--;
        }
        return true;
    }

    bool isStrictlyPalindromic(int n) {
        //between 2 -> n-2
        for(int i=2; i<=n-2; i++){
            //if not palindrom => return false
            string conv = bitset<64>(n).to_string();
            if(!checkPalindrom(conv)) return false;
        }
        return true;
    }
};

File: 2396. Strictly Palindromic Number/isStrictlyPalindromic.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isStrictlyPalindromic = function (n) {
    //between 2 -> n-2
    for (let i = 2; i <= n - 2; i++) {
        //if not palindrom => return FALSE
        let conv = n.toString(i)
        let rev = conv.split("").reverse().join("")
        if (conv !== rev) return false;
    }

    return true;
};